yarn.lock

```diff
@@ -2196,7 +2196,12 @@ chalk@2.4.2, chalk@^2.0.0, chalk@^2.4.1:
     escape-string-regexp "^1.0.5"
     supports-color "^5.3.0"
 
-chalk@4.1.2, chalk@^4.0.0, chalk@^4.1.1, chalk@^4.1.2:
+chalk@5.0.0:
+  version "5.0.0"
+  resolved "https://registry.yarnpkg.com/chalk/-/chalk-5.0.0.tgz#bd96c6bb8e02b96e08c0c3ee2a9d90e050c7b832"
+  integrity sha512-/duVOqst+luxCQRKEo4bNxinsOQtMP80ZYm7mMqzuh5PociNL0PvmHFvREJ9ueYL2TxlHjBcmLCdmocx9Vg+IQ==
+
+chalk@^4.0.0, chalk@^4.1.1, chalk@^4.1.2:
```

Comment:
I assume that chalk v4 comes from devtool sub-dependencies. Is that correct?

Reply:
I approved based on this assumption if the answer is yes and CI passes. The diff makes sense.

Reply:
Yes,

```console
>yarn why chalk
yarn why v1.22.17
[1/4] Why do we have the module "chalk"...?
[2/4] Initialising dependency graph...
[3/4] Finding dependency...
[4/4] Calculating file sizes...
=> Found "chalk@5.0.0"
info Has been hoisted to "chalk"
info This module exists because it's specified in "dependencies".
info Disk size without dependencies: "68KB"
info Disk size with unique dependencies: "68KB"
info Disk size with transitive dependencies: "68KB"
info Number of shared dependencies: 0
=> Found "eslint-formatter-friendly#chalk@2.4.2"
info This module exists because "eslint-formatter-friendly" depends on it.
info Disk size without dependencies: "40KB"
info Disk size with unique dependencies: "96KB"
info Disk size with transitive dependencies: "192KB"
info Number of shared dependencies: 6
=> Found "vnopts#chalk@2.4.2"
info This module exists because "vnopts" depends on it.
info Disk size without dependencies: "40KB"
info Disk size with unique dependencies: "96KB"
info Disk size with transitive dependencies: "192KB"
info Number of shared dependencies: 6
=> Found "npm-run-all#chalk@2.4.2"
info This module exists because "npm-run-all" depends on it.
info Disk size without dependencies: "40KB"
info Disk size with unique dependencies: "96KB"
info Disk size with transitive dependencies: "192KB"
info Number of shared dependencies: 6
=> Found "jest-diff#chalk@4.1.2"
info This module exists because "jest-diff" depends on it.
info Disk size without dependencies: "52KB"
info Disk size with unique dependencies: "88KB"
info Disk size with transitive dependencies: "184KB"
info Number of shared dependencies: 5
=> Found "jest-message-util#chalk@4.1.2"
info This module exists because "jest-message-util" depends on it.
info Disk size without dependencies: "52KB"
info Disk size with unique dependencies: "88KB"
info Disk size with transitive dependencies: "184KB"
info Number of shared dependencies: 5
=> Found "jest-snapshot#chalk@4.1.2"
info Reasons this module exists
   - "jest-snapshot" depends on it
   - Hoisted from "jest-snapshot#jest-diff#chalk"
info Disk size without dependencies: "52KB"
info Disk size with unique dependencies: "88KB"
info Disk size with transitive dependencies: "184KB"
info Number of shared dependencies: 5
=> Found "jest-util#chalk@4.1.2"
info This module exists because "jest-util" depends on it.
info Disk size without dependencies: "52KB"
info Disk size with unique dependencies: "88KB"
info Disk size with transitive dependencies: "184KB"
info Number of shared dependencies: 5
=> Found "babel-jest#chalk@4.1.2"
info This module exists because "babel-jest" depends on it.
info Disk size without dependencies: "52KB"
info Disk size with unique dependencies: "88KB"
info Disk size with transitive dependencies: "184KB"
info Number of shared dependencies: 5
=> Found "eslint#chalk@4.1.2"
info This module exists because "eslint" depends on it.
info Disk size without dependencies: "52KB"
info Disk size with unique dependencies: "88KB"
info Disk size with transitive dependencies: "184KB"
info Number of shared dependencies: 5
=> Found "jest-watch-typeahead#chalk@4.1.2"
info This module exists because "jest-watch-typeahead" depends on it.
info Disk size without dependencies: "52KB"
info Disk size with unique dependencies: "88KB"
info Disk size with transitive dependencies: "184KB"
info Number of shared dependencies: 5
=> Found "node-actionlint#chalk@4.1.2"
info This module exists because "node-actionlint" depends on it.
info Disk size without dependencies: "52KB"
info Disk size with unique dependencies: "88KB"
info Disk size with transitive dependencies: "184KB"
info Number of shared dependencies: 5
=> Found "cspell#chalk@4.1.2"
info This module exists because "cspell" depends on it.
info Disk size without dependencies: "52KB"
info Disk size with unique dependencies: "88KB"
info Disk size with transitive dependencies: "184KB"
info Number of shared dependencies: 5
=> Found "@babel/highlight#chalk@2.4.2"
info This module exists because "@babel#code-frame#@babel#highlight" depends on it.
info Disk size without dependencies: "40KB"
info Disk size with unique dependencies: "96KB"
info Disk size with transitive dependencies: "192KB"
info Number of shared dependencies: 6
=> Found "@jest/core#chalk@4.1.2"
info This module exists because "jest#@jest#core" depends on it.
info Disk size without dependencies: "52KB"
info Disk size with unique dependencies: "88KB"
info Disk size with transitive dependencies: "184KB"
info Number of shared dependencies: 5
=> Found "@jest/transform#chalk@4.1.2"
info This module exists because "jest-snapshot#@jest#transform" depends on it.
info Disk size without dependencies: "52KB"
info Disk size with unique dependencies: "88KB"
info Disk size with transitive dependencies: "184KB"
info Number of shared dependencies: 5
=> Found "@jest/types#chalk@4.1.2"
info This module exists because "@jest#globals#@jest#types" depends on it.
info Disk size without dependencies: "52KB"
info Disk size with unique dependencies: "88KB"
info Disk size with transitive dependencies: "184KB"
info Number of shared dependencies: 5
=> Found "jest-circus#chalk@4.1.2"
info This module exists because "@prettier#jest-light-runner#jest-circus" depends on it.
info Disk size without dependencies: "52KB"
info Disk size with unique dependencies: "88KB"
info Disk size with transitive dependencies: "184KB"
info Number of shared dependencies: 5
=> Found "jest-cli#chalk@4.1.2"
info This module exists because "jest#jest-cli" depends on it.
info Disk size without dependencies: "52KB"
info Disk size with unique dependencies: "88KB"
info Disk size with transitive dependencies: "184KB"
info Number of shared dependencies: 5
=> Found "jest-each#chalk@4.1.2"
info This module exists because "@prettier#jest-light-runner#jest-each" depends on it.
info Disk size without dependencies: "52KB"
info Disk size with unique dependencies: "88KB"
info Disk size with transitive dependencies: "184KB"
info Number of shared dependencies: 5
=> Found "jest-matcher-utils#chalk@4.1.2"
info Reasons this module exists
   - "@types#jest#jest-matcher-utils" depends on it
   - Hoisted from "@types#jest#jest-matcher-utils#jest-diff#chalk"
info Disk size without dependencies: "52KB"
info Disk size with unique dependencies: "88KB"
info Disk size with transitive dependencies: "184KB"
info Number of shared dependencies: 5
=> Found "snapshot-diff#chalk@4.1.2"
info Reasons this module exists
   - "snapshot-diff#jest-snapshot" depends on it
   - Hoisted from "snapshot-diff#jest-snapshot#chalk"
   - Hoisted from "snapshot-diff#jest-snapshot#jest-message-util#chalk"
   - Hoisted from "snapshot-diff#jest-snapshot#jest-util#chalk"
info Disk size without dependencies: "52KB"
info Disk size with unique dependencies: "88KB"
info Disk size with transitive dependencies: "184KB"
info Number of shared dependencies: 5
=> Found "jest-watcher#chalk@4.1.2"
info This module exists because "jest-watch-typeahead#jest-watcher" depends on it.
info Disk size without dependencies: "52KB"
info Disk size with unique dependencies: "88KB"
info Disk size with transitive dependencies: "184KB"
info Number of shared dependencies: 5
=> Found "@jest/console#chalk@4.1.2"
info This module exists because "jest#@jest#core#@jest#console" depends on it.
info Disk size without dependencies: "52KB"
info Disk size with unique dependencies: "88KB"
info Disk size with transitive dependencies: "184KB"
info Number of shared dependencies: 5
=> Found "@jest/reporters#chalk@4.1.2"
info This module exists because "jest#@jest#core#@jest#reporters" depends on it.info Disk size without dependencies: "52KB"
info Disk size with unique dependencies: "88KB"
info Disk size with transitive dependencies: "184KB"
info Number of shared dependencies: 5
=> Found "jest-config#chalk@4.1.2"
info This module exists because "jest#@jest#core#jest-config" depends on it.
info Disk size without dependencies: "52KB"
info Disk size with unique dependencies: "88KB"
info Disk size with transitive dependencies: "184KB"
info Number of shared dependencies: 5
=> Found "jest-resolve#chalk@4.1.2"
info This module exists because "jest#@jest#core#jest-resolve" depends on it.
info Disk size without dependencies: "52KB"
info Disk size with unique dependencies: "88KB"
info Disk size with transitive dependencies: "184KB"
info Number of shared dependencies: 5
=> Found "jest-runner#chalk@4.1.2"
info This module exists because "jest#@jest#core#jest-runner" depends on it.
info Disk size without dependencies: "52KB"
info Disk size with unique dependencies: "88KB"
info Disk size with transitive dependencies: "184KB"
info Number of shared dependencies: 5
=> Found "jest-runtime#chalk@4.1.2"
info This module exists because "jest#@jest#core#jest-runtime" depends on it.
info Disk size without dependencies: "52KB"
info Disk size with unique dependencies: "88KB"
info Disk size with transitive dependencies: "184KB"
info Number of shared dependencies: 5
=> Found "jest-validate#chalk@4.1.2"
info This module exists because "jest#@jest#core#jest-validate" depends on it.
info Disk size without dependencies: "52KB"
info Disk size with unique dependencies: "88KB"
info Disk size with transitive dependencies: "184KB"
info Number of shared dependencies: 5
=> Found "jest-jasmine2#chalk@4.1.2"
info This module exists because "jest#@jest#core#jest-config#jest-jasmine2" depends on it.
info Disk size without dependencies: "52KB"
info Disk size with unique dependencies: "88KB"
info Disk size with transitive dependencies: "184KB"
info Number of shared dependencies: 5
```

"vnopts#chalk@2.4.2" and "@babel/highlight#chalk@2.4.2" are removed in build script.

https://github.com/prettier/prettier/pull/12162 https://github.com/prettier/prettier/pull/12182